etna/datasets/tsdataset.py

```diff
@@ -240,6 +251,18 @@ def _prepare_df(cls, df: pd.DataFrame, freq: Optional[str]) -> pd.DataFrame:
 
         return df
 
+    @classmethod
+    def _prepare_df_exog(cls, df_exog: pd.DataFrame, freq: Optional[str]) -> pd.DataFrame:
+        df_format = DataFrameFormat.determine(df_exog)
+        if df_format is DataFrameFormat.long:
+            df_exog = cls.to_dataset(df_exog)
+
+        df_exog = cls._cast_segment_to_str(df=df_exog.copy(deep=True))
```

Comment:
Why here we redefine `df_exog` after `cls._cast_segment_to_str` and in `_prepare_df` method we use something like inplace?

Reply:
- We are making copy beforer calling `_prepare_df_exog`
- Assigning smth into a variable `df_exog` just shadows original value and it remains intact

I also think we could remove copying here, because we make copy before calling `_prepare_df_exog`.